### DISCUSSION TITLE
How do I enforce a new paragraph within a need directive?

### DISCUSSION THREAD

**Discussion by jumbo172** [useblocks/sphinx-needs#1278] [General]
Outside of a need directive, the following text renders like I want it, with each blank line separating the paragraphs:

```
Here is a paragraph followed by a single blank line.

**Procedure**

Here is another, preceded by a single blank line.
```

If I do the same inside a need directive, a blank line renders as a line break, not as a blank line to separate the paragraphs:

```
.. test_case::
    :id: example
    Here is a paragraph that is not followed by a single blank line.

    **Procedure**

    Here is another, that is not preceded by a single blank line.
```

How do I enforce blank lines to separate paragraphs, to aid readability?

**Comment by chrisjsewell** [comment]
Heya, it works the same as standard admonitions (like `note`), so will already do this.
Perhaps it is because you missed the blank line between the option (`:id:`) and the paragraph; with this I checked and it works fine

**Reply by jumbo172** [reply]
Hi Chris, thanks for the reply.

Here is my exact code:

```
.. test_case:: example
   :id: example

   Here is a paragraph that is not followed by a single blank line.

   **Procedure**

   Here is another, that is not preceded by a single blank line.
```
   
   It renders like this:
   
![need_paragrahp](https://github.com/user-attachments/assets/78bd52ff-7adb-4235-89a1-3487ab0a62b4)

What does yours render as?

Sphinx                         7.4.7
sphinx-needs               2.1.0
sphinx-rtd-theme        2.0.0

**Reply by chrisjsewell** [reply]
what does the actual html look like though? are they separate `<p>`
it is probably just CSS pushing them together

Note there has been changes to CSS in v3.0.0 so I would advise to move to that https://sphinx-needs.readthedocs.io/en/latest/changelog.html#documentation-and-css-styling

**Reply by jumbo172** [reply]
In the html each line is a separate `<p>`.

I have upgraded to v3.0.0 but I still see the same behaviour.

Inside an admonition, such as `note`, the lines are also separate `<p>`, but blank lines are rendered to separate the paragraphs.

**Reply by chrisjsewell** [reply]
> In the html each line is a separate `<p>`.

That would appear to be a CSS issue then.
What theme are you using? can you link to an example page, or use the https://developer.chrome.com/docs/devtools yourself to see what styling is being applied

**Reply by jumbo172** [reply]
I'm using `sphinx-rtd-theme` 2.0.0.

I used the devtools and inspected the formatting.

If I change `margin-bottom` from its default setting of 0 to 24, the line spacing renders as I would expect it to:

```
html.writer-html5 .rst-content table.docutils td>p,html.writer-html5 .rst-content table.docutils th>p {
    line-height: 1rem;
    margin-bottom: 24;
    font-size: .9rem
}
```

![html_margin](https://github.com/user-attachments/assets/1856a8b3-9902-4358-b589-7db853690388)

We are using a custom CSS, but I cant see anything in it that would be causing the problem.

**Comment by mcejp** [comment]
FWIW, I am experiencing the same with spring-rtd-theme and no custom CSS.
